### DISCUSSION TITLE
MapValue alternative on certain fields for bools

### DISCUSSION THREAD

**Discussion by Metadorius** [linq2db/linq2db#4422] [Q&A] [answered]
The problem: I am working on legacy database with some weird storage format decisions. It may be normal to have `NULL` correspond to `false`, and `1` as short to `true` on one field, and for example `NULL` = `false` and `"1"` (string of length 1) = `true` on another. Hence why the question if it's possible to have an attribute to map this weirdness to a proper `bool` (and/or maybe `bool?`). Is there something in Linq2DB arsenal to deal with this issue (declaratively)?

**Comment by sdanyliv** [comment, marked as answer]
It is possible.
Check this test https://github.com/linq2db/linq2db/blob/master/Tests/Linq/Linq/ValueConversionTests.cs#L800, `BoolConverterNullsAttribute` has conversion code. It is a just sample. 

Also it can be defined for field via `MappingSchema` using `HasConversion`

```cs
builder.Entity<MainClass>()
	.Property(e => e.BoolField)
	.HasConversion((bool? b) => b == true ? 1 : 0, m => m == 1 , true);
```
